OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -719,6 +719,65 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 					}
 				}
 
+				if (engineVersion < 20150314)
+				{
+					// Moved Reloads functionality to LimitedAmmo and refactored the latter into AmmoPool
+					if (depth == 0)
+					{
+						var actorTraits = node.Value.Nodes;
+						var limitedAmmo = actorTraits.FirstOrDefault(l => l.Key == "LimitedAmmo");
+						var reloads = actorTraits.FirstOrDefault(r => r.Key == "Reloads");
+
+						if (reloads != null)
+						{
+							var reloadsFields = reloads.Value.Nodes;
+							var limitedAmmoFields = limitedAmmo.Value.Nodes;
+							var count = reloadsFields.FirstOrDefault(c => c.Key == "Count");
+							var period = reloadsFields.FirstOrDefault(p => p.Key == "Period");
+							var resets = reloadsFields.FirstOrDefault(res => res.Key == "ResetOnFire");
+
+							var reloadsCount = count != null ? FieldLoader.GetValue<int>("Count", count.Value.Value) : -1;
+							var reloadsPeriod = period != null ? FieldLoader.GetValue<int>("Period", period.Value.Value) : 50;
+							var reloadsResetOnFire = resets != null ? FieldLoader.GetValue<bool>("ResetOnFire", resets.Value.Value) : false;
+
+							limitedAmmoFields.Add(new MiniYamlNode("SelfReloads", "true"));
+							limitedAmmoFields.Add(new MiniYamlNode("ReloadCount", reloadsCount.ToString()));
+							limitedAmmoFields.Add(new MiniYamlNode("SelfReloadTicks", reloadsPeriod.ToString()));
+							limitedAmmoFields.Add(new MiniYamlNode("ResetOnFire", reloadsResetOnFire.ToString()));
+
+							node.Value.Nodes.RemoveAll(n => n.Key == "Reloads");
+							node.Value.Nodes.RemoveAll(n => n.Key == "-Reloads");
+						}
+					}
+
+					// Moved RearmSound from Minelayer to LimitedAmmo/AmmoPool
+					if (depth == 0)
+					{
+						var actorTraits = node.Value.Nodes;
+						var limitedAmmo = actorTraits.FirstOrDefault(la => la.Key == "LimitedAmmo");
+						var minelayer = actorTraits.FirstOrDefault(ml => ml.Key == "Minelayer");
+
+						if (minelayer != null)
+						{
+							var minelayerFields = minelayer.Value.Nodes;
+							var limitedAmmoFields = limitedAmmo.Value.Nodes;
+							var rearmSound = minelayerFields.FirstOrDefault(rs => rs.Key == "RearmSound");
+
+							if (rearmSound != null)
```

Comment:
This should not be necessary.
Even if it's `null`, you still want to add "RearmSound" to `AmmoPool`.
Also on the next line you do a proper check.